### PR TITLE
#0 - refactored codes to use services and utils

### PR DESCRIPTION
**Pull request by NG-Solennel**
### What does this PR do?

Code refactoring to use services and utils properly

### Description of Task to be completed?

Separating controllers from services and utilities

### How should this be manually tested?

N/A

### Any background context you want to provide?

N/A

### What are the relevant pivotal tracker stories?

N/A

### Questions:

# Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code is formatted correctly
- [x] My changes doesn't have any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Ntarecedrick — APPROVED** [review]

**Review by Marthatwesi — APPROVED** [review]

**Review by Rutarenzi — APPROVED** [review]

File: src/controllers/userController.js [hunk 1]
```diff
@@ -1,58 +1,25 @@
...
-    return res.status(200).json({
-      name: req.user.name,
-      token,
-    });
-  } catch (error) {
-    return res.status(500).json({ message: "server error" });
+  static async login(req, res) {
+    try {
+      const response = await UserServices.login(req.user);
+      return res.status(200).json({ user: response });
```
Comment (Eli250) [inline src/controllers/userController.js:18]: The response should always return the message and status

File: src/controllers/userController.js [hunk 2]
```diff
@@ -1,58 +1,25 @@
...
-        username: user.name,
-        useremail: user.email,
-        UserToken: givenToken,
-      };
-      return res.status(201).send(userObj);
+      const response = await UserServices.register(req.body);
+      return res.status(201).send(response);
     } catch (error) {
+      console.log(error);
       return res.status(500).json({ error });
```
Comment (Eli250) [inline src/controllers/userController.js:11]: Always return the structured message and status

File: src/controllers/userController.js [hunk 3]
```diff
@@ -1,58 +1,25 @@
...
-        process.env.JWT_SECRET // eslint-disable-line
-      );
-      const userObj = {
-        username: user.name,
-        useremail: user.email,
-        UserToken: givenToken,
-      };
-      return res.status(201).send(userObj);
+      const response = await UserServices.register(req.body);
+      return res.status(201).send(response);
```
Comment (Eli250) [inline src/controllers/userController.js:8]: Your response should always be in json format and contains the message and status

File: src/middlewares/alreadyExists.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const { User } = require("../database/models");
+
+const alreadyExists = async (req, res, next) => {
```
Comment (Eli250) [inline src/middlewares/alreadyExists.js:3]: Rename your middleware to "verifyUser.js"

File: src/middlewares/alreadyExists.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  const exists = await User.findOne({ where: { email: req.body.email } });
+  if (exists) {
+    return res.status(409).send("Email Already Exists");
```
Comment (Eli250) [inline src/middlewares/alreadyExists.js:6]: The response should be in json format with both status and formatted message

File: src/middlewares/index.js [hunk 6]
```diff
@@ -1 +0,0 @@
-// middleware function belong here
```
Comment (Eli250) [inline src/middlewares/index.js:1]: If  you no longer need this file get rid of it.

File: src/middlewares/passport.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+import passport from "passport";
```
Comment (Eli250) [inline src/middlewares/passport.js:1]: This is a utility component, not a middleware. Move it to utils directory and call it in your controller

File: src/routes/api/user.route.js [hunk 8]
```diff
@@ -1,12 +1,18 @@
...
-import userVald from "../../validation/user.vald";
+import loginValidation from "../../validations/login.validation";
 
 const router = Router();
-router.post("/signup", signupValidation, UserController.register);
-router.post("/login", userVald, passport.authenticate("local"), userLogin);
+router.post(
+  "/signup",
+  signupValidation,
+  alreadyExists,
```
Comment (Eli250) [inline src/routes/api/user.route.js:13]: Rename this to verifyUser

File: src/utils/index.js [hunk 9]
```diff
@@ -1 +0,0 @@
-// files with dependencies to use
```
Comment (Eli250) [inline src/utils/index.js:1]: Get rid of this file if you no longer need it.

**Review by Eli250 — CHANGES_REQUESTED** [review]
Resolve comments

**Review by Eli250 — APPROVED** [review]

**Pull request merged**